tests/Main.hs

```diff
@@ -411,10 +456,15 @@ main = do
           proc
             "consul"
             [ "agent", "-dev"
+            , "-node", (unpack localhost) -- hardcode node name as "localhost"
             , "-log-level", "err"
+          --, "-log-level", "debug"        -- for debugging
             , "-http-port", show (fromIntegral consulPort :: Int)
             , "-config-file", configFilePath
             ]
     withProcessTerm consulProc $ \_p -> do
       waitForConsulOrFail
+      -- to let the consul agent register itself (the node the agent is running on)
+      -- TODO: should we instead query consul to lookup the node registration?
```

Comment:
Eventually most likely yes; "some seconds sleep" are always racy, and the best test suites are those that have none at all.

So I think you can go straight for

```
-- TODO: instead of racy sleeping query consul to lookup the node registration
```

Reply:
I considered it, but then backed off. Maybe you can help me find a route through this one:

The "obvious" way I could see to do this involves creating a client and using one of the API functions to see if the Node has been registered, and I wasn't sure if that was a reasonable thing to do as part of "setting up" the tests.